In [99]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [100]:
df = pd.read_csv('C:/Users/butle/hp/higher_ed/data/raw/Accountability_Metrics_Data.csv')

In [101]:
df.shape

(3206, 18)

In [102]:
df.info

<bound method DataFrame.info of       opeid6      level             sector  hbcu  completion_rate_150  \
0       1002  Four-year      Public 4-year     1               0.2685   
1       1003  Four-year   Nonprofit 4-year     0               0.2794   
2       1004  Four-year      Public 4-year     0               0.4917   
3       1005  Four-year      Public 4-year     1               0.3001   
4       1007   Two-year      Public 2-year     0               0.2832   
...      ...        ...                ...   ...                  ...   
3201   39803  Four-year      Public 4-year     0               0.5478   
3202   40513  Four-year  For-profit 4-year     0               0.2376   
3203   42527  Four-year   Nonprofit 4-year     0               0.2857   
3204   42542  Four-year   Nonprofit 4-year     0               0.4510   
3205   42544   Two-year      Public 2-year     0               0.4410   

      share_outstanding_ug_5  cdr3_wgtd  pct25_earn_wne_p10  est_fte  \
0                  

In [103]:
df.head()

opeid6      level            sector  hbcu  completion_rate_150  \
0    1002  Four-year     Public 4-year     1               0.2685   
1    1003  Four-year  Nonprofit 4-year     0               0.2794   
2    1004  Four-year     Public 4-year     0               0.4917   
3    1005  Four-year     Public 4-year     1               0.3001   
4    1007   Two-year     Public 2-year     0               0.2832   

   share_outstanding_ug_5  cdr3_wgtd  pct25_earn_wne_p10  est_fte  \
0                1.295501   0.136555             17800.0     5943   
1                1.182663   0.074850             21300.0     2702   
2                1.064660   0.041668             20400.0     2369   
3                1.307392   0.159253             15600.0     4234   
4                1.154031   0.041238             16100.0     1165   

   white_share  black_share  hisp_share  asian_share  aian_share  nhpi_share  \
0       0.0186       0.9120      0.0088       0.0018      0.0022      0.0016   
1       0.4238       0.4810      0.0178       0.0053      0.0024      0.0038   
2       0.6726       0.1666      0.0542       0.0107      0.0036      0.0018   
3       0.0167       0.9352      0.0095       0.0041      0.0013      0.0005   
4       0.7250       0.2337      0.0239       0.0041      0.0041      0.0000   

   twomore_share  nra_share  unkn_share  
0         0.0118     0.0070      0.0361  
1         0.0221     0.0241      0.0197  
2         0.0391     0.0275      0.0240  
3         0.0102     0.0102      0.0123  
4         0.0058     0.0017      0.0017

In [104]:
df.isnull().count()

opeid6                    3206
level                     3206
sector                    3206
hbcu                      3206
completion_rate_150       3206
share_outstanding_ug_5    3206
cdr3_wgtd                 3206
pct25_earn_wne_p10        3206
est_fte                   3206
white_share               3206
black_share               3206
hisp_share                3206
asian_share               3206
aian_share                3206
nhpi_share                3206
twomore_share             3206
nra_share                 3206
unkn_share                3206
dtype: int64

No null data

In [105]:
df.describe()

opeid6         hbcu  completion_rate_150  share_outstanding_ug_5  \
count   3206.000000  3206.000000          3206.000000             3206.000000   
mean    8706.120399     0.028072             0.500448                0.885001   
std     9628.070991     0.165205             0.218681                0.297760   
min     1002.000000     0.000000             0.000000                0.000000   
25%     2325.250000     0.000000             0.316725                0.819014   
50%     3668.500000     0.000000             0.496457                0.963213   
75%    10490.500000     0.000000             0.667550                1.064649   
max    42544.000000     1.000000             1.000000                1.332038   

         cdr3_wgtd  pct25_earn_wne_p10        est_fte  white_share  \
count  3206.000000         3206.000000    3206.000000  3206.000000   
mean      0.041727        21499.750249    4784.785402     0.519438   
std       0.042737         9135.721459    8233.290886     0.254534   
min       0.000000         3200.000000       9.000000     0.000000   
25%       0.011597        14900.000000     763.250000     0.343334   
50%       0.030206        19350.000000    1980.500000     0.569048   
75%       0.054870        27100.000000    5163.250000     0.720870   
max       0.314423        81900.000000  101635.000000     1.000000   

       black_share   hisp_share  asian_share   aian_share   nhpi_share  \
count  3206.000000  3206.000000  3206.000000  3206.000000  3206.000000   
mean      0.156962     0.169025     0.039733     0.014108     0.003948   
std       0.188248     0.204581     0.070646     0.073191     0.028947   
min       0.000000     0.000000     0.000000     0.000000     0.000000   
25%       0.040800     0.044800     0.007700     0.001500     0.000000   
50%       0.088781     0.091450     0.018000     0.003316     0.001100   
75%       0.194075     0.206475     0.040500     0.006816     0.002700   
max       1.000000     1.000000     0.990700     1.000000     0.994300   

       twomore_share    nra_share   unkn_share  
count    3206.000000  3206.000000  3206.000000  
mean        0.034302     0.026167     0.036319  
std         0.032444     0.048437     0.058239  
min         0.000000     0.000000     0.000000  
25%         0.017000     0.000000     0.005100  
50%         0.030659     0.009300     0.018900  
75%         0.043834     0.030779     0.043000  
max         0.456800     0.702691     0.747300

Do students who identify as black and or as two or more races complete college more often and have higher earnings than students who identify as black at non-HBCU private and public schools? 

In [106]:
HBCU = df[df['hbcu'] == 1]
HBCU.head()

opeid6      level            sector  hbcu  completion_rate_150  \
0     1002  Four-year     Public 4-year     1               0.2685   
3     1005  Four-year     Public 4-year     1               0.3001   
10    1017   Two-year     Public 2-year     1               0.2133   
17    1028  Four-year  Nonprofit 4-year     1               0.2421   
19    1030   Two-year     Public 2-year     1               0.2166   

    share_outstanding_ug_5  cdr3_wgtd  pct25_earn_wne_p10  est_fte  \
0                 1.295501   0.136555             17800.0     5943   
3                 1.307392   0.159253             15600.0     4234   
10                0.000000   0.000000             14100.0     3644   
17                1.301007   0.102731             15300.0     1604   
19                0.000000   0.000000             13600.0     2209   

    white_share  black_share  hisp_share  asian_share  aian_share  nhpi_share  \
0        0.0186       0.9120      0.0088       0.0018      0.0022      0.0016   
3        0.0167       0.9352      0.0095       0.0041      0.0013      0.0005   
10       0.7056       0.1905      0.0388       0.0063      0.0058      0.0007   
17       0.0330       0.9483      0.0078       0.0000      0.0006      0.0000   
19       0.2418       0.6635      0.0246       0.0246      0.0074      0.0008   

    twomore_share  nra_share  unkn_share  
0          0.0118     0.0070      0.0361  
3          0.0102     0.0102      0.0123  
10         0.0294     0.0219      0.0010  
17         0.0097     0.0000      0.0006  
19         0.0164     0.0164      0.0045

Is opeid6 1017 really and HBCU? Its share of black-identifying students is .19 and white-identifying students is .70

In [107]:
HBCU.shape

(90, 18)

What is the cutoff I should use for HBCUs?

In [108]:
poss = HBCU[HBCU['black_share'] < .50].count()
poss

opeid6                    6
level                     6
sector                    6
hbcu                      6
completion_rate_150       6
share_outstanding_ug_5    6
cdr3_wgtd                 6
pct25_earn_wne_p10        6
est_fte                   6
white_share               6
black_share               6
hisp_share                6
asian_share               6
aian_share                6
nhpi_share                6
twomore_share             6
nra_share                 6
unkn_share                6
dtype: int64

In [109]:
poss = HBCU[HBCU['black_share'] < .50]
poss.head(6)

opeid6      level         sector  hbcu  completion_rate_150  \
10      1017   Two-year  Public 2-year     1               0.2133   
1559    3608   Two-year  Public 2-year     1               0.2068   
1668    3764  Four-year  Public 4-year     1               0.3916   
1710    3809  Four-year  Public 4-year     1               0.2600   
1722    3826  Four-year  Public 4-year     1               0.2923   
1978    5691   Two-year  Public 2-year     1               0.2237   

      share_outstanding_ug_5  cdr3_wgtd  pct25_earn_wne_p10  est_fte  \
10                  0.000000   0.000000             14100.0     3644   
1559                1.119589   0.005676             16100.0     6429   
1668                1.244651   0.118788             21200.0     4236   
1710                1.153602   0.104576             14200.0     1104   
1722                1.168624   0.050273             16500.0     2401   
1978                0.000000   0.000000             15000.0     3430   

      white_share  black_share  hisp_share  asian_share  aian_share  \
10         0.7056       0.1905      0.0388       0.0063      0.0058   
1559       0.1970       0.1135      0.6229       0.0236      0.0034   
1668       0.0045       0.2451      0.0033       0.0013      0.0015   
1710       0.8614       0.0721      0.0135       0.0071      0.0024   
1722       0.6901       0.1432      0.0185       0.0056      0.0129   
1978       0.5503       0.3748      0.0016       0.0065      0.0034   

      nhpi_share  twomore_share  nra_share  unkn_share  
10        0.0007         0.0294     0.0219      0.0010  
1559      0.0010         0.0307     0.0016      0.0064  
1668      0.0000         0.0000     0.0125      0.7318  
1710      0.0000         0.0285     0.0063      0.0087  
1722      0.0028         0.0747     0.0157      0.0365  
1978      0.0010         0.0240     0.0047      0.0337

In [110]:
poss2 = HBCU[HBCU['black_share'] < .75]
poss2.shape

(24, 18)

In [111]:
poss2

opeid6      level            sector  hbcu  completion_rate_150  \
10      1017   Two-year     Public 2-year     1               0.2133   
19      1030   Two-year     Public 2-year     1               0.2166   
257     1428  Four-year     Public 4-year     1               0.4008   
264     1441  Four-year     Public 4-year     1               0.2353   
287     1486  Four-year  Nonprofit 4-year     1               0.3614   
324     1544  Four-year     Public 4-year     1               0.1924   
587     1968  Four-year     Public 4-year     1               0.1798   
626     2026  Four-year     Public 4-year     1               0.2500   
672     2106  Four-year     Public 4-year     1               0.3855   
891     2479  Four-year     Public 4-year     1               0.2146   
1148    2926  Four-year     Public 4-year     1               0.4245   
1150    2928  Four-year     Public 4-year     1               0.3533   
1177    2962  Four-year  Nonprofit 4-year     1               0.1535   
1274    3157  Four-year     Public 4-year     1               0.1291   
1499    3522  Four-year     Public 4-year     1               0.3006   
1536    3577  Four-year  Nonprofit 4-year     1               0.2330   
1559    3608   Two-year     Public 2-year     1               0.2068   
1668    3764  Four-year     Public 4-year     1               0.3916   
1710    3809  Four-year     Public 4-year     1               0.2600   
1722    3826  Four-year     Public 4-year     1               0.2923   
1770    3946  Four-year     Public 4-year     1               0.2305   
1892    5260   Two-year     Public 2-year     1               0.1382   
1978    5691   Two-year     Public 2-year     1               0.2237   
1985    5734   Two-year     Public 2-year     1               0.2782   

      share_outstanding_ug_5  cdr3_wgtd  pct25_earn_wne_p10  est_fte  \
10                  0.000000   0.000000             14100.0     3644   
19                  0.000000   0.000000             13600.0     2209   
257                 1.217076   0.089352             21500.0     4318   
264                 1.128706   0.056475             16800.0     3168   
287                 1.328768   0.104650             14500.0     1227   
324                 1.247588   0.086317             19500.0     5348   
587                 1.293366   0.075554             16400.0     1440   
626                 1.277672   0.088414             16400.0     1962   
672                 1.232434   0.084814             22400.0     2663   
891                 1.185108   0.171133             18800.0     2087   
1148                1.200652   0.103309             16900.0     1557   
1150                1.198775   0.073904             18900.0     5566   
1177                1.303029   0.184884             16600.0     1317   
1274                1.282571   0.158537             15200.0     2015   
1499                1.241139   0.094260             21800.0     6505   
1536                1.252618   0.131472             18100.0      959   
1559                1.119589   0.005676             16100.0     6429   
1668                1.244651   0.118788             21200.0     4236   
1710                1.153602   0.104576             14200.0     1104   
1722                1.168624   0.050273             16500.0     2401   
1770                1.059217   0.041569             19800.0     1635   
1892                0.000000   0.000000             12600.0      566   
1978                0.000000   0.000000             15000.0     3430   
1985                0.000000   0.000000             13900.0     1419   

      white_share  black_share  hisp_share  asian_share  aian_share  \
10         0.7056       0.1905      0.0388       0.0063      0.0058   
19         0.2418       0.6635      0.0246       0.0246      0.0074   
257        0.0832       0.7392      0.0694       0.0052      0.0026   
264        0.0424       0.6042      0.1307       0.0189      0.0025   
287        0.0072       0.6670      0.0797       

It looks like a black-student percentage of above 50 is a good cutoff to consider for my HBCUs

In [126]:
df.columns[n_HBCU] = HBCU[HBCU['black_share'] < .50] | df[df['hbcu'] == 0]

TypeError: Index does not support mutable operations

In [113]:
non_hbcu = df[df['hbcu'] != 1] 
non_hbcu.shape

(3116, 18)

In [114]:
not_hbcu = pd.concat([non_hbcu, n_HBCU])

In [115]:
not_hbcu.shape

(3122, 18)

In [116]:
HBCU.shape

(90, 18)

In [117]:
HBCU.head()

opeid6      level            sector  hbcu  completion_rate_150  \
0     1002  Four-year     Public 4-year     1               0.2685   
3     1005  Four-year     Public 4-year     1               0.3001   
10    1017   Two-year     Public 2-year     1               0.2133   
17    1028  Four-year  Nonprofit 4-year     1               0.2421   
19    1030   Two-year     Public 2-year     1               0.2166   

    share_outstanding_ug_5  cdr3_wgtd  pct25_earn_wne_p10  est_fte  \
0                 1.295501   0.136555             17800.0     5943   
3                 1.307392   0.159253             15600.0     4234   
10                0.000000   0.000000             14100.0     3644   
17                1.301007   0.102731             15300.0     1604   
19                0.000000   0.000000             13600.0     2209   

    white_share  black_share  hisp_share  asian_share  aian_share  nhpi_share  \
0        0.0186       0.9120      0.0088       0.0018      0.0022      0.0016   
3        0.0167       0.9352      0.0095       0.0041      0.0013      0.0005   
10       0.7056       0.1905      0.0388       0.0063      0.0058      0.0007   
17       0.0330       0.9483      0.0078       0.0000      0.0006      0.0000   
19       0.2418       0.6635      0.0246       0.0246      0.0074      0.0008   

    twomore_share  nra_share  unkn_share  
0          0.0118     0.0070      0.0361  
3          0.0102     0.0102      0.0123  
10         0.0294     0.0219      0.0010  
17         0.0097     0.0000      0.0006  
19         0.0164     0.0164      0.0045

In [118]:
to_drop = HBCU[HBCU['black_share'] < .5].index.tolist()
to_drop

[10, 1559, 1668, 1710, 1722, 1978]

In [119]:
HBCU = HBCU.drop(to_drop)
HBCU.shape

(84, 18)

Now I have the two dfs I want: HBCU which contains the HBCUs that have greater than or equal to 50% black students, and not_HBCU, which contains all the other schools

In [124]:
from statistics import mean
ave_HBCU_com_rate = mean(HBCU['completion_rate_150'])
ave_HBCU_com_rate

0.3269261904642857